game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCombatValue.java

```diff
@@ -0,0 +1,95 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AaDefenseCombatValue implements CombatValue {
+
+  @NonNull GameData data;
+  @NonNull AvailableSupports friendlySupportTracker;
```

Comment:
Is 'friendlySupportTracker' a legacy name when the 'supports' was a tracker? WDYT to rename these to 'friendlySupports' and 'enemySupports'

Also worth considering renaming 'enemySupports' to 'enemyNegativeBuffs'. 'enemySupports' sounds like we are tracking both defense and offense and the respective supports for each side. My understanding is the enemySupport here is an "anti-support" being imposed by enemy units (hence, a negative buff).

Reply:
I renamed it to `supportFromFriends` and `supportFromEnemies`.  I didn't want to use `enemyNegativeBuffs` because it might not be a negative buff.  I haven't seen a case where an enemy support rule improved a unit but I have seen a case where a friend support rule hurt a unit.

Reply:
Good point, I like the rename, it's more clear.